### PR TITLE
Update documentation index page and navigation structure

### LINKED ISSUE #2783: Docs landing page should showcase examples
This is just a suggestion based on user feedback.

The landing page of the docs (https://nest-simulator.readthedocs.org/) could showcase our coolest examples ("applications") from https://nest-simulator.readthedocs.io/en/v3.4/examples/index.html#pynest-examples, rather than a little script snippet which is typically glossed over (too much technical detail for a landing page). Applications are much more telling about what you can do with NEST (what it can do for you!)

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR improves the overall landing page and structure of contents. 

[See output here](https://nest-simulator--2969.org.readthedocs.build/en/2969/)

Resolves #2783 

## The new index page 

- It removes the index.html template in favour of a restructured text page, which will eliminate issues with compatibility between the sphinx_material theme and custom template. 
- contains a 'carousel' where we can highlight new and interesting examples. (TODO, decide on which examples to showcase)
- contains a modified table of contents (TOC): details below
- contains a conceptual diagram using elements from NEST Desktop and images courtesy of A. Fischer  TODO: Add description to diagram
- A 'get started here' button leads to the new tutorials and guides section, which is what is seen on the TOC 

## The new tutorials and guides section

As discussed with @clinssen, as well as others,  we decided to encapsulate all document material together in one place. Previously the tutorial for PyNEST and MUSIC were separated from other document guides. 

I chose to use heading _Tutorials and Guides_, because not all the documents we have can be defined as tutorials in my opinion.

The tutorials and guides are structured in cards similar to before, but now the first row is organized by topics that should help users get started:

* the first card simply is the first steps - getting to know NEST - here the installation instructions are repeated, the PyNEST tutorial is added, along with the one neuron example, for those who want a quicker overview, and the video tutorial for those who prefer that medium
* The second card deals with the actual creation of the script that the user needs, and what guides would be helpful for this. In this regard  I was thinking of the basic things you do with NEST (create, connect, simulate, record) note: stimulate is not included for brevity and we don't have that extensive amount of docs on it. In this context, I am making the assumption here that users would need to look into models, connection management, simulation behavior and recorders for getting data out for analysis.
* The third card deals more with the code aspects of the script, that is, the functionality allowed by Nodecollections, parameter objects. This would help users optimize their script and show them ways of manipulating the data for their desired outcomes.

After this first row, comes _Additional topics_ 
Here I try to organize them as they were before, with a bit more text is added to help clarify the links.

### Changes in the Networks card

Added to the `spatially_structured_networks `is the network models (mam and microcircuit). Not sure if this is the best choice.

## The main table of contents

Now the table of contents is divided into 3 sections "usage", "community" and "related projects"

* _Usage_ contains the basic topics for docs - install, tutorials and guides, examples, API, models, glossary and then model implementations (which could be removed if it doesnt belong on top level), and technical docs (renamed from developer docs to move away from user vs developer)
* _Community_ contains topics related to contributing to NEST, citing NEST, contact, what's new in NEST and also link to the homepage (we could also add publications and events as links within the homepage?). Basically all the things that involve the user community 
* _Related projects_ contain the other NEST projects - we discussed adding a diagram for other projects and explain the ecosystem that NEST is in, but that is for another PR

## Other changes

- A modified colour theme is used (removing the green in favour of sticking to orange-brown and light and dark blue)
-  A cleanup of the static folder, and included jquery into template layout.html which solved a bug in the search.
- The contribute section is split from the old developer docs in an attempt to make contribution section more available for everyone

### CODE REVIEW

**Comment by clinssen** [comment]
Thanks for the great work!

Just some early observatations:

I would not make "model implementations" a top-level category, at it just contains some scattered notes about a few models. Instead, make sure these notebooks are linked from the models to which they apply. For instance, link the Hill & Tononi notebook from the Hill & Tononi neuron and synapse model documentation pages.

I would move away from the "conceptual approach" imagemap as it's hard to navigate and the interface is a bit yesteryear. Instead, just expand all of the categories that can be clicked on the image into tiles (stimulation devices, neurons, synapses, connections, and recording devices). 

What is the difference between "technical documentation" and "contribute to NEST"? Suggest to merge and use only the latter heading.

**Comment by jessica-mitchell** [comment]
> Thanks for the great work!
> 
> Just some early observatations:
> 
> I would not make "model implementations" a top-level category, at it just contains some scattered notes about a few models. Instead, make sure these notebooks are linked from the models to which they apply. For instance, link the Hill & Tononi notebook from the Hill & Tononi neuron and synapse model documentation pages.

Yah, ok they should be  linked from respective models, just wanted to check

> I would move away from the "conceptual approach" imagemap as it's hard to navigate and the interface is a bit yesteryear. Instead, just expand all of the categories that can be clicked on the image into tiles (stimulation devices, neurons, synapses, connections, and recording devices).

I see your point;  @jougs do you  have thoughts on this, since this is an idea we discussed?  The goal here was to provide a visual representation of how a network comes together. 

> 
> What is the difference between "technical documentation" and "contribute to NEST"? Suggest to merge and use only the latter heading.

Right now, you're right there isnt much difference, but I am hoping we get some more technical docs that lean more on the C++ side.

File: doc/htmldoc/developer_space/index.rst [hunk 1]
```diff
@@ -17,76 +17,27 @@ It is documentation for anyone needing to touch the code or documentation.
...
+   .. grid-item-card:: Reviewer guidelines
 
-        If you are requested to review a pull request, please
-        check our :ref:`code_guidelines`
+       * If you are requested to review a pull request, please check our :ref:`code_guidelines`
 
+   .. grid-item-card:: CI Workflow
 
-    .. grid-item-card::  Continuous integration
+       * Here you can find details on our :ref:`CI workflow <cont_integration>`
```
Comment (clinssen) [inline doc/htmldoc/developer_space/index.rst:40]: The "SLI documentation" button just below here is devoid of text. Could we add a little one-liner? "SLI is the internal scripting language of NEST. It is now deprecated and not recommended for new applications". Perhaps also on the sli_docs index page as well.

File: doc/htmldoc/developer_space/workflows/documentation_workflow/developer_documentation_workflow.rst [hunk 2]
```diff
@@ -1,10 +1,10 @@
 .. _devdoc_workflow:
 
-Developer documentation workflow
-################################
+Techincal documentation workflow
```
Comment (clinssen) [inline doc/htmldoc/developer_space/workflows/documentation_workflow/developer_documentation_workflow.rst:3]:
```suggestion
Technical documentation workflow
```

File: doc/htmldoc/devices/index.rst [hunk 3]
```diff
@@ -1,10 +1,29 @@
 .. _device_index:
 
-All about devices in NEST
-=========================
+All about recording and stimulating networks in NEST
```
Comment (clinssen) [inline doc/htmldoc/devices/index.rst:3]: Please add a link to this document under "More about devices" on get-started_index.html — or alternatively, delete ``devices/index.rst`` page as it contains very little content; just merge the links in to the "More about devices" box on get-started_index.html.

File: doc/htmldoc/index.rst [hunk 4]
```diff
@@ -1,16 +1,139 @@
...
+
+
+.. grid::
+
+  .. grid-item::
+
+    NEST is used in computational neuroscience to model and study behavior of large networks of neurons.
+
+    The models describe single neuron and synapse behavior and their connections.
+    Different mechanisms of plasticity can be used to investigate artificial learning
```
Comment (clinssen) [inline doc/htmldoc/index.rst:16]:
```suggestion
    Different models of plasticity can be used to investigate the mechanisms learning
```
possibly, this is a "standard text" paragraph and edits should go in somewhere upstream instead?

File: doc/htmldoc/index.rst [hunk 5]
```diff
@@ -1,16 +1,139 @@
...
+             :ref-type: ref
+             :color: info
+             :align: center
+             :shadow:
+
+             Click here to discover all our examples!
+
+----
+
+Conceptual approach
```
Comment (clinssen) [inline doc/htmldoc/index.rst:68]: I would vote to remove this paragraph and the imagemap figure.

File: doc/htmldoc/static/img/arrow_left.svg
Comment (clinssen) [inline doc/htmldoc/static/img/arrow_left.svg:1]: there seems to be a glitch in this figure on the top

File: doc/htmldoc/static/img/device_text.svg
Comment (clinssen) [inline doc/htmldoc/static/img/device_text.svg:1]: issue with transparency

File: doc/htmldoc/static/img/synapse_text.svg
Comment (clinssen) [inline doc/htmldoc/static/img/synapse_text.svg:1]: Just to double-check that Angela is credited as original creator of the images.

**Review by clinssen — COMMENTED** [review]
Super cool and a huge improvement, much obliged!

**Comment by jessica-mitchell** [comment]
@clinssen I am closing this PR in lieu of splitting it up into multiple PRs for a cleaner and easier review process. Thanks for your contributions thus far.

**Pull request closed without merging**